django/db/backends/sqlite3/base.py

```diff
@@ -375,6 +377,8 @@ def _sqlite_date_trunc(lookup_type, dt):
 def _sqlite_time_trunc(lookup_type, dt):
     try:
         dt = backend_utils.typecast_time(dt)
+        if dt is None:
+            return None
```

Comment:
Move this part outside part outside of the `try` branch just like in `_sqlite_date_trunc`.